### ISSUE TITLE
QuPath 0.2.0 cannot open project files stored in folders containing special characters by "double click-MIME-type-open-with"

### ISSUE THREAD

**Issue by sekro** [qupath/qupath#320]
**You've checked https://qupath.github.io for a new release that might already have fixed the issue, right?**

**And you've searched for related issues here to see if the issues has already been reported?**

Good, then please fill in the template about the bug...

> For general questions about the software (not bugs) use [image.sc](https://forum.image.sc/tags/qupath)

----

**Describe the bug**
Opening a project file by double-clicking in file file explorer [Windows 10 x64] fails if path contains special characters [in my case it was Ø]. QuPath error message: Could not open file. Opening the same project file residing in the same folder by using QuPath menu option "Open project" works fine.

**To Reproduce**
On Windows 10 x64:
1. create a project in a folder containing a special character (e.g. ø)
2. double click the project file in file explorer to open through MIME-type association
3. QuPath opens but cannot open the project

**Expected behavior**
QuPath opens the project

**Screenshots**
not needed

**Desktop (please complete the following information):**
 - OS: Windows 10 x64
 - QuPath Version 0.2.0

**Additional context**
---

**Comment by petebankhead** [comment]
Thanks for the clear description, I can reproduce the problem but it's not clear to me where exactly the character is getting messed up. Consequently, I can't tell it's something that can be properly resolved within QuPath or not.

Specifically, if I launch QuPath from the command line passing a path containing ø then it works as it should. However, if I double-click on the same file, then somewhere the character becomes Ã˜. It appears that QuPath 'receives' the wrong version, and I don't know if this is thanks to Windows, jpackage or something else.

There is a known bug in jpackage (used to create the .exe) whereby the application cannot be started if it is in a path that contains a non-ASCII character, and this impacts QuPath: https://bugs.openjdk.java.net/browse/JDK-8205991

I suspect therefore this isn't a QuPath bug, but rather a jpackage-related one.

**Comment by petebankhead** [comment]
Note that it does appear possible to convert the string back, in that this works in a script:
```groovy
s = 'file:/SÃ˜mething.qpproj'
print s
s2 = s.getBytes()
s2 = new String(s2, 'UTF-8')
print s2
```
gives the result
```
INFO: file:/SÃ˜mething.qpproj
INFO: file:/SØmething.qpproj
```
However it feels a little bit scary to be making this change as I don't know what other trouble it might cause, and would prefer to rely on it being fixed as source.

For what it's worth, I never rely on double-clicking to open projects anyway... rather, I drag the .qpproj file (or the folder containing it) onto QuPath, or select from the _File &rarr; Recent projects... &rarr;_ menu.

**Comment by petebankhead** [comment]
Since there are these other options, and it doesn't _seem_ to be a bug within QuPath itself, my suggested approach is to do nothing and hope for a fix elsewhere...  but happy to discuss alternatives.

**Comment by sekro** [comment]
I see your point and agree - maybe you can add a warning in case a user chooses to save a project in a folder with a special character in its path?
Same goes for the error message when trying to open from a "dangerous" path.

**Comment by petebankhead** [comment]
I'm not sure how exactly to identify what characters are special in a cross-platform way... besides, I feel that a warning might be unnecessarily obtrusive, since as far as I can tell only double-clicking (on Windows?) is affected, but all the other methods of opening should work.

**Comment by sekro** [comment]
Assuming the reason is the bug you mentioned (https://bugs.openjdk.java.net/browse/JDK-8205991) you could filter for non ASCII and for OS - if Windows 10 and non-ASCII = true -> warn.
I do not think it is intrusive. Better warn the user and point him directly to a simple solution rather than dealing with annoyed and clueless users. Special characters in words and names are not so rare (especially in the Windows world ;) ) and for example in hospital settings the usage of Windows is quite common.

**Comment by petebankhead** [comment]
The pull request above contains a proposed fix.

Upon reflection, I'm not sure where exactly the encoding issue arises (Java, jpackage, Wix used to create the exe, the Windows command prompt... or somewhere between them) so I haven't been able to find any clear documentation of expected behavior.

I hope the suggested fix is sufficiently modest that it can fix most issues without creating new ones, but that remains to be tested. Feedback/fixes/suggestions welcome. If it looks basically ok, it will be included in v0.2.1.

**Comment by sekro** [comment]
Sounds great - I'll see if I can find some time to test it and report back :)

**Comment by petebankhead** [comment]
Closing this because the proposed fix is now available in v0.2.1. Feel free to reopen or create a new issue if a related problem remains.
